src/Database/Mirror.php

```
@@ -781,15 +781,20 @@ function ($doc) use ($onNext, &$modified) {
         return $modified;
     }
 
-    public function createOrUpdateDocuments(string $collection, array $documents, int $batchSize = Database::INSERT_BATCH_SIZE, callable|null $onNext = null): int
-    {
+    public function upsertDocuments(
+        string $collection,
+        array $documents,
+        int $batchSize = Database::INSERT_BATCH_SIZE,
+        ?callable $onNext = null,
+        ?callable $onError = null,
```

PHPMD (2.15.0)

Avoid unused parameters such as '$onError'. (Unused Code Rules)

(UnusedFormalParameter)